Zotero/Scenes/Detail/PDF/Views/PDFAnnotationsViewController.swift

```diff
@@ -164,113 +183,94 @@ final class PDFAnnotationsViewController: UIViewController {
 
-    /// Updates `UIImage` of `SquareAnnotation` preview if the cell is currently on screen.
-    /// - parameter keys: Set of keys to update.
-    private func updatePreviewsIfVisible(for keys: Set<String>) {
-        let cells = tableView.visibleCells.compactMap({ $0 as? AnnotationCell }).filter({ keys.contains($0.key) })
-
-        for cell in cells {
-            let image = viewModel.state.previewCache.object(forKey: (cell.key as NSString))
-            cell.updatePreview(image: image)
-        }
-    }
-
-    /// Reloads tableView if needed, based on new state. Calls completion either when reloading finished or when there was no reload.
-    /// - parameter state: Current state.
-    /// - parameter completion: Called after reload was performed or even if there was no reload.
-    private func reloadIfNeeded(for state: PDFReaderState, completion: @escaping () -> Void) {
-        if state.document.pageCount == 0 {
-            DDLogWarn("AnnotationsViewController: trying to reload empty document")
-            completion()
-            return
-        }
-
-        let isVisible = parentDelegate?.isSidebarVisible ?? false
-
-        if state.changes.contains(.annotations) {
-            if state.changes.contains(.sidebarEditing) {
-                tableView.setEditing(state.sidebarEditingEnabled, animated: false)
-            }
-            updateQueue.async { [weak self] in
-                guard let self else { return }
-                var snapshot = NSDiffableDataSourceSnapshot<Int, PDFReaderState.AnnotationKey>()
-                snapshot.appendSections([0])
-                snapshot.appendItems(state.sortedKeys)
-                if let keys = state.updatedAnnotationKeys {
-                    snapshot.reloadItems(keys)
-                }
-                dataSource.apply(snapshot, animatingDifferences: isVisible, completion: completion)
+        /// Reloads tableView if needed, based on new state. Calls completion either when reloading finished or when there was no reload.
+        /// - parameter state: Current state.
+        /// - parameter completion: Called after reload was performed or even if there was no reload.
+        func reloadIfNeeded(for state: PDFReaderState, completion: @escaping () -> Void) {
+            if state.document.pageCount == 0 {
+                DDLogWarn("AnnotationsViewController: trying to reload empty document")
+                completion()
+                return
             }
-            return
-        }
 
-        if state.changes.contains(.interfaceStyle) {
-            updateQueue.async { [weak self] in
-                guard let self else { return }
-                var snapshot = dataSource.snapshot()
-                guard !snapshot.sectionIdentifiers.isEmpty else { return }
-                snapshot.reloadSections([0])
-                dataSource.apply(snapshot, animatingDifferences: false, completion: completion)
+            let isVisible = parentDelegate?.isSidebarVisible ?? false
```

Comment:
This is already created "above" in `update(state: PDFReaderState)`, you can add it to `reloadIfNeeded` as parameter.

Reply:
Check https://github.com/zotero/zotero-ios/pull/1040/commits/19acd1e7612bb3e5dcc261eeb6f3c6f0b963d740